### PR TITLE
Get username in case of certificates authentication

### PR DESCRIPTION
**Pull request by nicoldr**
Signed-off-by: Nicol Draghici <idraghic@cisco.com>

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/config-tls.json [hunk 1]
```diff
@@ -9,7 +9,14 @@
     "realm": "zot",
     "tls": {
       "cert": "test/data/server.cert",
-      "key": "test/data/server.key"
+      "key": "test/data/server.key",
+      "cacert": "test/data/ca.crt"
```
Comment (peusebiu) [inline examples/config-tls.json:13]: remove this changes, no need for them.

File: pkg/api/authz.go [hunk 2]
```diff
@@ -194,7 +194,18 @@ func AuthzHandler(ctlr *Controller) mux.MiddlewareFunc {
...
 			/* To do: verify client certs and get its username(subject DN)
 			if request.TLS.VerifiedChains != nil, then get subject DN
-			issue: https: //github.com/project-zot/zot/issues/614 */
+			issue: https://github.com/project-zot/zot/issues/614 */
+
+			if request.TLS.VerifiedChains != nil {
+				
+				for _, cert:=range request.TLS.PeerCertificates {
+					acCtrlr.Log.Info().Msgf("Certificate elements: %#v", cert.Subject.CommonName)
+				}
```
Comment (peusebiu) [inline pkg/api/authz.go:203]: here we need to take the username from request.TLS.PeerCertificates or request.TLS.VerifiedChains

File: pkg/api/authz.go [hunk 3]
```diff
@@ -194,7 +194,18 @@ func AuthzHandler(ctlr *Controller) mux.MiddlewareFunc {
...
+
+				if err != nil {
```
Comment (peusebiu) [inline pkg/api/authz.go:205]: this err is always nil :D

File: pkg/api/controller_test.go [hunk 4]
```diff
@@ -971,6 +971,93 @@ func TestTLSWithBasicAuthAllowReadAccess(t *testing.T) {
 	})
 }
 
+func TestTLSWithDefaultUserPermission(t *testing.T) {
```
Comment (peusebiu) [inline pkg/api/controller_test.go:974]: in this test we need to test that tls username works, not anonymous.
we need a certificate with non nil CN or subjectDN and put that into user policy.

ex *.mydomain.com
```
				AuthorizationAllRepos: config.PolicyGroup{
					Policies: []config.Policy{
						{
							Users:   []string{"*.mydomain.com"},
							Actions: []string{"read"},
						},
					}
```

**Comment by peusebiu** [comment]
https://github.com/hashicorp/vault/pull/2595

File: examples/config-tls.json (hunk 1 above)
Comment (nicoldr) [reply examples/config-tls.json:13]: Will remove them when making the PR

**Pull request closed without merging**